src/macaron/slsa_analyzer/git_url.py

```diff
@@ -376,6 +377,45 @@ def clone_remote_repo(clone_dir: str, url: str) -> Repo | None:
     return Repo(path=clone_dir)
 
 
+def list_remote_references(arguments: list[str], repo: str) -> str | None:
+    """Retrieve references from a remote repository using Git's ``ls-remote``.
+
+    Parameters
+    ----------
+    arguments: list[str]
+        The arguments to pass into the command.
+    repo: str
+        The repository to run the command on.
+
+    Returns
+    -------
+    str
+        The result of the command.
+    """
+    try:
+        result = subprocess.run(  # nosec B603
+            args=["git", "ls-remote"] + arguments + [repo],
+            capture_output=True,
+            # By setting stdin to /dev/null and using a new session, we prevent all possible user input prompts.
+            stdin=subprocess.DEVNULL,
+            start_new_session=True,
+            cwd=global_config.output_path,
+            check=False,
+        )
+    except (subprocess.CalledProcessError, OSError):
+        return None
+
+    if result.returncode != 0:
+        error_string = result.stderr.decode("utf-8").strip()
+        if error_string.startswith("fatal: could not read Username"):
+            # Occurs when a repository cannot be accesses either because it does not exist, or it requires a login
+            # that is blocked.
+            logger.error("Could not access repository: %s", repo)
+        return None
```

Comment:
It would be good to also print a general error message when the error message does not start with `fatal: could not read Username`.